### PR TITLE
tracker: change `tidb_enable_change_column_type` default value to ON

### LINKED ISSUE #1769: schema tracker should set `tidb_enable_change_column_type = true`
## Feature Request

**Is your feature request related to a problem? Please describe:**

https://asktug.com/t/topic/93630

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 
close #1769

### What is changed and how it works?
as title

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes
- tracker's defaultGlobalVars change

Side effects
- allow change dll colum type

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: tests/shardddl4_1/run.sh [hunk 1]
```diff
@@ -667,9 +668,18 @@ function DM_150_CASE {
 	run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,\"bbbbbbb\");"
 	run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,\"ccccccc\");"
 
-	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
-		"query-status test" \
-		"Unsupported modify column: length 10 is less than origin 20" 1
+	if [[ "$shardmode" == "pessimistic" ]]; then
+		# modify column a double ddl passed but is still waiting for reslove from the sharding group
```
Comment (lance6716) [inline tests/shardddl4_1/run.sh:672]: I can't read this sentence

File: tests/shardddl4_1/run.sh [hunk 2]
```diff
@@ -667,9 +668,18 @@ function DM_150_CASE {
...
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+			"query-status test" \
+			'"target": "`shardddl`.`tb`"', 2
```
Comment (lance6716) [inline tests/shardddl4_1/run.sh:675]:
```suggestion
			'"target": "`shardddl`.`tb`"' 2
```

Comment (lance6716) [reply tests/shardddl4_1/run.sh:675]: and rest of them

File: tests/shardddl4_1/run.sh [hunk 3]
```diff
@@ -667,9 +668,18 @@ function DM_150_CASE {
 	run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,\"bbbbbbb\");"
 	run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,\"ccccccc\");"
 
-	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
-		"query-status test" \
-		"Unsupported modify column: length 10 is less than origin 20" 1
+	if [[ "$shardmode" == "pessimistic" ]]; then
+		# modify column a double ddl passed but is still waiting for reslove from the sharding group
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+			"query-status test" \
```
Comment (lance6716) [inline tests/shardddl4_1/run.sh:674]: check a more clear message, like ddl-lock / unsynced ddl

File: tests/shardddl4_1/run.sh [hunk 4]
```diff
@@ -696,21 +707,31 @@ function DM_151_CASE {
...
+
 }
 
 function DM_151 {
-	run_case 151 "double-source-pessimistic" \
-		"run_sql_source1 \"create table ${shardddl1}.${tb1} (id int primary key, a int);\"; \
-         run_sql_source2 \"create table ${shardddl1}.${tb1} (id int primary key, a int);\"; \
-         run_sql_source2 \"create table ${shardddl1}.${tb2} (id int primary key, a int);\"" \
-		"clean_table" "pessimistic"
+	# run_case 151 "double-source-pessimistic" \
```
Comment (lance6716) [inline tests/shardddl4_1/run.sh:725]: why

Comment (Ehco1996) [reply tests/shardddl4_1/run.sh]: Forgot to uncomment after testing locally - -!

File: tests/shardddl4_1/run.sh [hunk 5]
```diff
@@ -696,21 +711,36 @@ function DM_151_CASE {
...
+		# ddl: "modify column a double;" passes in worker1, but in pessimistic mode is still waiting for the other worker in the sharding group to be executed with the same ddl.
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+			"query-status test" \
+			'"ALTER TABLE `shardddl`.`tb` MODIFY COLUMN `a` DOUBLE"' 2
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+			"show-ddl-locks" \
+			'"ALTER TABLE `shardddl`.`tb` MODIFY COLUMN `a` DOUBLE"' 1
+	else
+		# ddl: "modify column a double" is passed in optimistic mode and will be executed downstream.
+		# but changing the int column to a double column is not allowed,so task is paused
```
Comment (lance6716) [inline tests/shardddl4_1/run.sh:724]:
```suggestion
		# but changing the int column to a double column is not allowed, so task is paused
```

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
test failed

**Comment by Ehco1996** [comment]
> test failed
@lance6716 

after many times retry, ci finally fassed

I don't know why this happens (it fails in different case every time).

File: Makefile [hunk 6]
```diff
@@ -172,6 +172,15 @@ dm_integration_test_build: tools_setup
 	$(FAILPOINT_DISABLE)
 	tests/prepare_tools.sh
 
+dm_integration_test_build_worker: tools_setup
```
Comment (GMHDBJD) [inline Makefile:175]: It is a bit strange if there is only `dm_integration_test_build_worker` but no `dm_integration_test_build_master` 😂

File: tests/_utils/handle_error_lib.sh [hunk 7]
```diff
@@ -40,7 +40,7 @@ function clean_table() {
 	run_sql_tidb "drop table if exists ${db}.${ta};"
 	run_sql_tidb "drop table if exists ${db}.${tb1};"
 	run_sql_tidb "drop table if exists ${db}.${tb2};"
-
+	run_sql_both_source "reset master"
```
Comment (GMHDBJD) [inline tests/_utils/handle_error_lib.sh:43]: Any problem without this line?

File: tests/shardddl4_1/run.sh [hunk 8]
```diff
@@ -696,21 +710,36 @@ function DM_151_CASE {
 	run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5);"
 	run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6);"
 
-	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
-		"query-status test" \
-		"Unsupported modify column: type double not match origin int(11), and tidb_enable_change_column_type is false" 1
+	if [[ "$shardmode" == "pessimistic" ]]; then
```
Comment (GMHDBJD) [inline tests/shardddl4_1/run.sh:713]: If this case now is supported, we may change it to a positive case, and use check_sync_diff below.

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: Makefile (hunk 6 above)
Comment (Ehco1996) [reply Makefile:175]: it's convenient for local test, if someone often needs to change/build dm-master , he can add `dm_integration_test_build_master` 😂

File: tests/_utils/handle_error_lib.sh (hunk 7 above)
Comment (Ehco1996) [reply tests/_utils/handle_error_lib.sh:43]: It is helpful when debugging binlog files manually

File: tests/shardddl4_1/run.sh (hunk 8 above)
Comment (Ehco1996) [reply tests/shardddl4_1/run.sh:718]: added

File: tests/_utils/handle_error_lib.sh (hunk 7 above)
Comment (GMHDBJD) [reply tests/_utils/handle_error_lib.sh:43]: But `reset master` rarely occurs upstream. Users mostly start a task with the previous binlog files. I think test without `reset master` cover more scenes

File: tests/shardddl4_1/run.sh [hunk 9]
```diff
@@ -667,17 +668,34 @@ function DM_150_CASE {
...
+			"show-ddl-locks" \
+			'ALTER TABLE `shardddl`.`tb` MODIFY COLUMN `a` VARCHAR(10)"' 1
+
+		# we alter database in source2 and the ddl lock will be resolved
+		run_sql_source2 "alter table ${shardddl1}.${tb1} modify column a varchar(10);"
+		run_sql_source2 "alter table ${shardddl1}.${tb2} modify column a varchar(10);"
+		check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+	else
+		# ddl: "modify column a varchar(10)" is passed in optimistic mode and will be executed downstream.
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
```
Comment (GMHDBJD) [inline tests/shardddl4_1/run.sh:686]: Optimistic can also use check_sync_diff?

Comment (Ehco1996) [reply tests/shardddl4_1/run.sh:686]: Optimistic mode does not pass, the task has been paused

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**